web/META-INF/resources/js/controllers.js

```diff
@@ -0,0 +1,280 @@
+            else
+                opt = $rootScope.searchOptions.opt;
+
+            SearchOptions.setSearchOptions({'query':word, 'opt':opt});
+            $rootScope.searchOptions.query = searchKeyWords;
+            $rootScope.searchOptions.opt = opt;
+            searchMetadata();
+            //opt = 'And';
+        }
+
+        function initController() {
+            // initialize to page 1
+            vm.setPage(1);
+        }
+
+        function setPage(page) {
+            if (page < 1 || page > vm.pager.totalPages) {
+                return;
+            }
+
+            // get pager object from service
+            vm.pager = PagerService.GetPager(vm.PDItems.length, page);
+
+            // get current page of items
+            vm.items = vm.PDItems.slice(vm.pager.startIndex, vm.pager.endIndex + 1);
+        }
+        
+        function rankData(opt){
+        	rankopt = opt;
+        	searchMetadata();
+        }
+
+        this.searchTopic = function(topic) {
+            var searchKeyWords = new String();
+
+            if (topic.search(',') != -1 ) {
+                var topics = topic.split(',');
+                for (var i = 0; i < topics.length; i++) {
+                    searchKeyWords += topics[i];
+                }
+            } else {
+                searchKeyWords = topic;
+            } 
+            
+            if (searchKeyWords.search('/') != -1) {
+                var topics = searchKeyWords.split('/');
+                searchKeyWords = "";
+                for (var i = 0; i < topics.length; i++) {
+                    searchKeyWords += topics[i];
+                }
+            } 
+            
+            $rootScope.searchOptions.query = searchKeyWords;
+            MetaData.get({query: searchKeyWords, operator: opt}, 
+                function success(response) {
+                    vm.PDItems = response.PDResults;
+                    vm.totalMatches = vm.PDItems.length;
+                    vm.query = word;
+                    vm.opt = opt;
+                    initController();
+                    //$scope.PDResults = response.PDResults;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+            );
+        }
+  
+        function searchMetadata(){
+        	MetaData.get({query: word, operator: opt, rankoption: rankopt}, 
+                    function success(response) {
+                        vm.PDItems = response.PDResults;
+                        vm.totalMatches = vm.PDItems.length;
+                        vm.query = word;
+                        vm.opt = opt;
+                        initController();
+                        //$scope.PDResults = response.PDResults;
+                    },
+                    function error(errorResponse) {
+                        console.log("Error:" + JSON.stringify(errorResponse));
+                    }
+            );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "opt" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a literal value for this. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3APrimitiveWrappers)